src/languages/gml.js

```diff
@@ -1,18 +1,55 @@
 /*
 Language: GML
-Description: Game Maker Language for GameMaker (rev. 2023.1)
-Website: https://manual.yoyogames.com/
+Description: Game Maker Language for GameMaker (rev. 2024.11)
+Website: https://manual.gamemaker.io/
 Category: scripting
 */
 
 export default function(hljs) {
+
+  /**
+   * @type {any[][][]}
+   */
+  const lateConcatTargets = [];
+
+  /**
+   * Somewhat evil helper function to allow us to *very* lately build arrays. This is most useful
+   * in the case of {@link EXPRESSION}, which is an array of a bunch of different types, of which
+   * some of these also can contain expressions, AND contain other things too.
+   * 
+   * Note that {@link doLateConcat} MUST be called before the results of this are valid, because
+   * we're using some reference-based trickery to do this.
+   * 
+   * @param  {...any[]} arrays The arrays to be concatenated later.
+   * @returns {any[]}
+   */
+  function lateConcat(...arrays) {
+    lateConcatTargets.push(arrays);
+    return arrays;
+  }
+
+  /**
+   * Perform the late-concatenation operation on all target arrays, converting them into effectively
+   * an array of the members that were to be concatenated.
+   */
+  function doLateConcat() {
+
```

Comment:
Could you explain why these helpers are necessary?

Reply:
Effectively, they're a workaround for `EXPRESSION` being a big array of "everything that is an expression", of which some expression types (e.g., template strings) can, themselves, contain expressions. I saw that some other languages handled this by juggling `thing.contains.push(...)` but this felt pretty unwieldy, so that's what I came up with to work around it.

Though, I actually only ended up using it in one place in the current setup, and that's somewhere that it can be removed from, so I might go ahead and take it out, perhaps revisit if need be in the future? Dunno. I'll remove it. :)